### ISSUE TITLE
shape inferrence about dynamic axis

### ISSUE THREAD

**Issue by ChipmunkKeeper** [onnx/onnx#6477]
I'm trying to export the pytorch model into onnx without dynamic axis and infer the shape with shape_inference. But the inferred model still have unkown dimensions. How to ensure all the dimensions in onnx is fixed? Thank you.

![微信截图_20241021124448](https://github.com/user-attachments/assets/1cea6db5-3b8b-4220-8123-ea0461fb5cc2)
![微信截图_20241021125032](https://github.com/user-attachments/assets/85a20849-f20d-4e3e-a74e-f7938a4e5ade)

The code  is as follows:
```
model = torchvision.models.detection.retinanet_resnet50_fpn(pretrained=True)
model.eval()  
dummy_input = torch.randn(1, 3, 800, 800)
onnx_model_path = "retinanet.onnx"
torch.onnx.export(model, 
                  dummy_input, 
                  onnx_model_path, 
                  export_params=True,  
                  opset_version=12,    
                  do_constant_folding=True, 
                  input_names=['input'],  
                  output_names=['output'],
            ) 
model = onnx.load(onnx_model_path)
inferred_model = onnx.shape_inference.infer_shapes(model)
onnx.save(inferred_model, onnx_model_path)
```

**Comment by xadupre** [comment]
Is it possible to know the shape used to reshape? onnx is not able to handle expression, a tensor with dimensions [a,b] reshaped into (-1,) leaves the shape unknown even though it should be a*b. Your model does not have dynamic axis but the shape you are sharing does have unknown value. They behave like dynamic ones. Is it possible to check where these appeared?

**Comment by ChipmunkKeeper** [comment]
I find it appears after the operator "Resize". Some values of the tensor is empty and it's ouput is unkown. Maybe it is the problem. Do you know how to solve it?

![微信截图_20241021172124](https://github.com/user-attachments/assets/a15afc97-e9d5-4149-8465-9899fab82bd8)

**Comment by xadupre** [comment]
shape inference for Resize op probably fails because the data about scales is not available (probably saved as external data), same goes for roi. Maybe you can load the model with the data and save it again while keeping those in the model with external weights. How do you use the shapes after that?

**Comment by ChipmunkKeeper** [comment]
That doesn't work and I think there's somthing wrong in the operator reshape. I can successfully infer other models without reshape.

**Comment by ChipmunkKeeper** [comment]
I see. Thank you for your help!
